Fixed replaced image in product page

NoZTurn opened the pull request:
with this small code trick ,we can change the replaced image to the newest one immediately the time after the image ajax replaced without the page refresh or reload.

<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | with this small code trick ,we can change the replaced image to the newest one immediately the time after the image ajax replaced without the page refresh or reload.
| Type?             | improvement 
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #20035.
| How to test?      | Just clear your caches and refresh your product v2 page.
| Possible impacts? | small change, no other impact.


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/24983)
<!-- Reviewable:end -->

NoZTurn committed:
Update Dropzone.vue

with this small code trick ,we can change the replaced image to the newest one immediately the time after the image ajax replaced with the page refresh or reload.

NeOMakinG commented on admin-dev/themes/new-theme/js/pages/product/components/dropzone/Dropzone.vue:
```diff
@@ -494,7 +494,7 @@
             DropzoneMap.savedImage(newImage.image_id),
           );
-          imageElement.src = newImage.image_url;
+          imageElement.src = newImage.image_url + '?' + Math.random();
```
What's the behavior behind this? It's because of browser's cache?

NoZTurn replied:
yes, I think is the browser cache, you can force the browser to renew the new image url after the image uploaded.

NeOMakinG replied:
@PierreRambaud wdyt? I really appreciate this little trick

PierreRambaud replied:
If we want to do something properly we have to get the image checksum instead of a random number.
Using `Math.random` can return twice the same number.

NoZTurn replied:
checksum is the best solution, if Math.random() may return a repeat number, we can also use new Date() and its getTime() function instead of Math.random(), because all times a different, so this can avoid the same number repeat.

PierreRambaud replied:
If the image is only load once, why not, otherwise, if it has to be always displayed, we should use the checksum method

NeOMakinG replied:
> If the image is only load once, why not, otherwise, if it has to be always displayed, we should use the checksum method

Looks like it doesn't have to be always displayed, it's only on replace, in order to kick the browser's cache out which is faster, but maybe every API endpoints can just provide the URL with image checksum : ping @jolelievre

NoZTurn replied:
> > If the image is only load once, why not, otherwise, if it has to be always displayed, we should use the checksum method
> 
> Looks like it doesn't have to be always displayed, it's only on replace, in order to kick the browser's cache out which is faster, but maybe every API endpoints can just provide the URL with image checksum : ping @jolelievre

agree with you, I come across that at product list page, the image is also not updated after image replacement because of the browser's cache, so maybe we have to makes some change at the image core API.

NeOMakinG commented:
@PrestaShop/prestashop-core-developers As this PR is not a bug but an improvement, do we want it to target the 1.7.8.x branch? It's fast to merge and improve the new product page, but the new build is really close :/

NoZTurn committed:
Update Dropzone.vue

checksum is the best solution,but us a different tail number is more simple, if Math.random() may return a repeat number, we can also use new Date() and its getTime() function instead of Math.random(), because all times a different, so this can avoid the same number repeat.

I renew my pull request above.

PierreRambaud commented on admin-dev/themes/new-theme/js/pages/product/components/dropzone/Dropzone.vue:
```diff
@@ -494,7 +494,8 @@
             DropzoneMap.savedImage(newImage.image_id),
           );
-          imageElement.src = newImage.image_url;
+          var imageUpdateTime = new Date();
```
```suggestion
          const imageUpdateTime = new Date();
```

NoZTurn committed:
Update admin-dev/themes/new-theme/js/pages/product/components/dropzone/Dropzone.vue

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

NeOMakinG commented on admin-dev/themes/new-theme/js/pages/product/components/dropzone/Dropzone.vue:
```diff
@@ -494,7 +494,7 @@
           );
-          var imageUpdateTime = new Date();
+          const imageUpdateTime = new Date();
           imageElement.src = newImage.image_url + '?' + imageUpdateTime.getTime();
```
```suggestion
          imageElement.src = `${newImage.image_url}?${imageUpdateTime.getTime()}`;
```

Damn, I don't now why eslint didn't cry before

Commit:
Change concatenation to literals

NeOMakinG commented:
@MatShir so we need to focus develop on this PR?

MatShir commented:
@Progi1984 @NeOMakinG, friendly reminder, in a beta phase to avoid new bugs and to stabilize the software, we don't accept improvements. Could you please transfer the contribution into develop please ?
And the improvement is applied to the new product page, which is hidden on the feature flag, so I can wait for the next version :)

@NoZTurn, thanks a lot for the improvements. It is very appreciated, but it is not on the right branch 😄

NeOMakinG commented:
@NoZTurn we need to do these changes on the develop branch.

The rebase is a bit hardcore, so I would suggest you to cherry pick commits into a new PR based on develop, don't hesitate if you need help. I could do it, but it's your work and I don't wan't to steal it :)

NoZTurn commented:
@MatShir @NeOMakinG thanks for your sugguestion, and I have submit this pull request to the develop branch here https://github.com/PrestaShop/PrestaShop/pull/25089.

NeOMakinG commented:
Closing in favor of #25089. Thanks @NoZTurn
